openpype/hosts/nuke/plugins/publish/extract_slate_frame.py

```diff
@@ -29,145 +34,165 @@ class ExtractSlateFrame(openpype.api.Extractor):
         "f_vfx_scope_of_work": [False, ""]
     }
 
+    # constants
+    SLATE_TO_SEQUENCE_DONE = None
+
     def process(self, instance):
-        if hasattr(self, "viewer_lut_raw"):
-            self.viewer_lut_raw = self.viewer_lut_raw
-        else:
-            self.viewer_lut_raw = False
+        self.fpath = instance.data["path"]
+        self.first_frame = instance.data["frameStartHandle"]
+        self.last_frame = instance.data["frameEndHandle"]
+
+        self.log.info("Creating staging dir...")
+
+        if "representations" not in instance.data:
+            instance.data["representations"] = []
+
+        self._create_staging_dir(instance)
 
         with maintained_selection():
             self.log.debug("instance: {}".format(instance))
             self.log.debug("instance.data[families]: {}".format(
                 instance.data["families"]))
 
-            self.render_slate(instance)
-
-    def render_slate(self, instance):
-        node_subset_name = instance.data.get("name", None)
-        node = instance[0]  # group node
-        self.log.info("Creating staging dir...")
-
-        if "representations" not in instance.data:
-            instance.data["representations"] = list()
-
+            if instance.data.get("bakePresets"):
+                for o_name, o_data in instance.data["bakePresets"].items():
+                    self.log.info("_ o_name: {}, o_data: {}".format(
+                        o_name, pformat(o_data)))
+                    self.render_slate(instance, o_name, **o_data)
+            else:
+                viewer_process_swithes = {
+                    "bake_viewer_process": True,
+                    "bake_viewer_input_process": True
+                }
+                self.render_slate(instance, None, **viewer_process_swithes)
+
+    def _create_staging_dir(self, instance):
         staging_dir = os.path.normpath(
-            os.path.dirname(instance.data['path']))
+            os.path.dirname(self.fpath))
 
         instance.data["stagingDir"] = staging_dir
 
         self.log.info(
             "StagingDir `{0}`...".format(instance.data["stagingDir"]))
 
-        frame_start = instance.data["frameStart"]
-        frame_end = instance.data["frameEnd"]
-        handle_start = instance.data["handleStart"]
-        handle_end = instance.data["handleEnd"]
+    def render_slate(self, instance, output_name=None, **kwargs):
```

Comment:
Why there are not mentioned required arguments?
```suggestion
    def render_slate(
        self,
        instance,
        bake_viewer_process,
        bake_viewer_input_process,
        output_name=None
    ):
```